src/sentry/api/bases/organization_events.py

```diff
@@ -290,6 +293,15 @@ def build_cursor_link(self, request: Request, name: str, cursor: Optional[Cursor
             has_results="true" if bool(cursor) else "false",
         )
 
+    def handle_on_demand(self, request: Request) -> tuple[bool, MetricSpecType]:
```

Comment:
A docstring would be useful.

Maybe rename it to "determine_if_to_use_on_demand_metrics_and_metric_type".